client/src/views/workspaces/WorkspaceHiddenModal.vue

```diff
@@ -0,0 +1,102 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ion-page class="modal">
+    <ms-modal
+      title="WorkspacesPage.workspaceHiddenModal.title"
+      :close-button="{ visible: true }"
+      :confirm-button="{
+        disabled: workspaceName === undefined,
+        label: 'WorkspacesPage.workspaceHiddenModal.actionConfirm',
+        onClick: confirmHidden,
+      }"
+      :cancel-button="{
+        label: 'WorkspacesPage.workspaceHiddenModal.actionCancel',
+        disabled: false,
+        onClick: cancel,
+      }"
+    >
+      <div class="workspace-hidden">
+        <ion-text class="workspace-hidden__description body-lg">
+          <i18n-t
+            keypath="WorkspacesPage.workspaceHiddenModal.affectedWorkspaces"
+            scope="global"
+          >
+            <template #workspace>
+              <strong>{{ workspaceName ? workspaceName : '' }}</strong>
+            </template>
+          </i18n-t>
+        </ion-text>
+        <ms-report-text :theme="MsReportTheme.Info">
+          {{ $msTranslate('WorkspacesPage.workspaceHiddenModal.info') }}
+        </ms-report-text>
+      </div>
+      <ms-checkbox
+        label-placement="end"
+        class="workspace-hidden__checkbox body"
+        v-model="skipWorkspaceHiddenWarning"
+      >
+        <ion-text>
+          {{ $msTranslate('WorkspacesPage.workspaceHiddenModal.noReminder') }}
+        </ion-text>
+      </ms-checkbox>
+    </ms-modal>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import { WorkspaceName } from '@/parsec';
+import { IonPage, IonText, modalController } from '@ionic/vue';
+import { MsCheckbox, MsModal, MsModalResult, MsReportText, MsReportTheme } from 'megashark-lib';
+import { ref } from 'vue';
+
+const props = defineProps<{
+  workspaceName: WorkspaceName;
+}>();
+
+const skipWorkspaceHiddenWarning = ref(false);
+
+async function confirmHidden(): Promise<boolean> {
+  if (!props.workspaceName) {
```

Comment:
Cannot happen, workspaceName is not declared as optional. Well technically it can happen if the name is empty but that's the caller's fault.